src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -250,10 +302,43 @@ protected virtual IEnumerable<ServiceBusPlugin> DefineServiceBusPlugins()
         /// <param name="cancellationToken">Indicates that the shutdown process should no longer be graceful.</param>
         public override async Task StopAsync(CancellationToken cancellationToken)
         {
+            if ((Settings.Options.TopicSubscription & TopicSubscription.DeleteOnStop) == TopicSubscription.DeleteOnStop)
```

Comment:
Is this a shorthand notation of `Settings.Options.TopicSubscription != null && Settings.Options.TopicSubscription == TopicSubscription.DeleteOnStop`?

Reply:
The enum is actually just a number (so can't be `null`). I added the `[Flags]` attribute to indicate that we expect combinations of the enum values.

So, the notation is a bitwise condition to filter out first the `.DeleteOnStop` value (which is `3` from the configured options value (`1` for example).
Bitwise is `3`, `0011` and `1`, `0001`. The `&` filters out all other bits. Like if we would have set the options to `7` (`0111`) it would become `0011`.
This final value (`0011`) we can use to match against the thing we actually want to check `3`.

So, in short, the syntax allows you to first filter out ("clean" if you will) the bits that would make up the expected value, and then check if it's indeed the expected value.
You follow?

Reply:
Can't we just use `Settings.Options.TopicSubscription.HasFlag(TopicSubscription.DeleteOnStop)` which does what you want to achieve?

Reply:
Ah, yes, that would make the code maybe more expressive, but the really downside on that approach is that it has a unnecessary boxing/unboxing on the integer bc it also checks the type of the enum. Bitwise operations are much faster here.
So, ... IDK. I can change it if you like. There's not that much to win I guess.

Reply:
Let's just follow the typical .NET approach by using `HasFlag`